tsdb/head.go

```diff
@@ -1539,9 +1539,14 @@ func (h *headIndexReader) Symbols() index.StringIter {
 	return index.NewStringListIter(res)
 }
 
-// LabelValues returns the possible label values
+// LabelValues returns label values present in the head for the
+// specific label name that are within the time range mint to maxt.
 func (h *headIndexReader) LabelValues(name string) ([]string, error) {
 	h.head.symMtx.RLock()
+	if h.maxt < h.head.MinTime() || h.mint > h.head.MaxTime() {
+		return []string{}, nil
```

Comment:
`h.head.symMtx.RUnlock()` before returning :) Not in a `defer` above because we don't want `sort.Strings(sl)` inside the lock.

Optional and unrelated to this PR: I see that `LabelNames()` has the unlock in the defer, can you change it to be similar to this method too?